File: src/services/tauri/panel.ts
import { invoke } from '@tauri-apps/api/core'

import type { AimValue, BotItem, Difficulty, LaunchResult, NadesValue, PanelInitializationResult, PanelMode, PanelSnapshot } from '@/features/panel/types'

export const initializePanelDefaults = (rootPath: string) => invoke<PanelInitializationResult>('initialize_panel_defaults', { rootPath })
export const getPanelSnapshot = (rootPath: string) => invoke<PanelSnapshot>('get_panel_snapshot', { rootPath })
export const setPanelMode = (rootPath: string, mode: PanelMode) => invoke<PanelSnapshot>('set_panel_mode', { rootPath, mode })
export const setPanelDifficulty = (rootPath: string, level: Difficulty) => invoke<PanelSnapshot>('set_panel_difficulty', { rootPath, level })
export const setPanelAim = (rootPath: string, value: AimValue) => invoke<PanelSnapshot>('set_panel_aim', { rootPath, value })
export const setPanelNades = (rootPath: string, value: NadesValue) => invoke<PanelSnapshot>('set_panel_nades', { rootPath, value })
export const setPanelBotItem = (rootPath: string, item: BotItem, enabled: boolean) => invoke<PanelSnapshot>('set_panel_bot_item', { rootPath, item, enabled })
export const setPanelDropKnives = (rootPath: string, bindKey: string, selected: number[]) => invoke<PanelSnapshot>('set_panel_drop_knives', { rootPath, bindKey, selected })
export const launchPanelCs2 = (rootPath: string, mode: PanelMode) => invoke<LaunchResult>('launch_panel_cs2', { rootPath, mode })

File: src-tauri/src/commands/panel.rs
use crate::errors::AppError;
use crate::models::panel::{LaunchResult, PanelInitializationResult, PanelSnapshot};
use crate::services::panel;
use tauri::AppHandle;

#[tauri::command]
pub fn get_panel_snapshot(root_path: String) -> Result<PanelSnapshot, String> {
    panel::snapshot(&root_path).map_err(AppError::into_string)
}

#[tauri::command]
pub fn initialize_panel_defaults(root_path: String) -> Result<PanelInitializationResult, String> {
    panel::initialize_panel_defaults(&root_path).map_err(AppError::into_string)
}

#[tauri::command]
pub fn set_panel_mode(
    app: AppHandle,
    root_path: String,
    mode: String,
) -> Result<PanelSnapshot, String> {
    panel::set_mode_with_app(&app, &root_path, &mode).map_err(AppError::into_string)
}

#[tauri::command]
pub fn set_panel_difficulty(root_path: String, level: String) -> Result<PanelSnapshot, String> {
    panel::set_difficulty(&root_path, &level).map_err(AppError::into_string)
}

#[tauri::command]
pub fn set_panel_aim(root_path: String, value: String) -> Result<PanelSnapshot, String> {
    panel::set_preset(&root_path, "bot_aim", &value).map_err(AppError::into_string)
}

#[tauri::command]
pub fn set_panel_nades(root_path: String, value: String) -> Result<PanelSnapshot, String> {
    panel::set_preset(&root_path, "bot_nades", &value).map_err(AppError::into_string)
}

#[tauri::command]
pub fn set_panel_bot_item(
    root_path: String,
    item: String,
    enabled: bool,
) -> Result<PanelSnapshot, String> {
    panel::set_bot_item(&root_path, &item, enabled).map_err(AppError::into_string)
}

#[tauri::command]
pub fn set_panel_drop_knives(
    root_path: String,
    bind_key: String,
    selected: Vec<u16>,
) -> Result<PanelSnapshot, String> {
    panel::set_drop_knives(&root_path, &bind_key, &selected).map_err(AppError::into_string)
}

#[tauri::command]
pub fn launch_panel_cs2(
    app: AppHandle,
    root_path: String,
    mode: String,
) -> Result<LaunchResult, String> {
    panel::launch_cs2(&app, &root_path, &mode).map_err(AppError::into_string)
}
